lib/translation.rb

```diff
@@ -45,15 +49,29 @@ def configure(&block)
         # include is private until Ruby 2.1
         Proxy.send(:include, GetText)
 
-        Proxy.bindtextdomain(TEXT_DOMAIN, {
-          :path           => @config.locales_path,
-          :output_charset => @config.charset
-        })
+        if @config.multi_domain
+          # TODO: Here we iterate over the langauges to initialize them
+          @config.domain_names.each do |domain|
+            Proxy.bindtextdomain(domain, {
+              path: @config.locales_path,
+              output_charset: @config.charset
+              })
+          end
```

Comment:
Conditional block based on new conflig-flag
Iterate over each text domain specified and bind to gettext.
My understanding is this creates a lookup chain of translation domains.  
It might be worth investigating if ordering domains affects performance, i.e. should main app be added first or last in chain?